test/catch.cc

```diff
@@ -86,12 +86,39 @@ TEST(CatchTest, All) {
                  ADD_FAILURE() << "Encountered unexpected matched raised";
                  return 10;
                })
-               .all([](std::exception_ptr&& error) {
-                 try {
-                   std::rethrow_exception(error);
-                 } catch (const std::runtime_error& error) {
-                   EXPECT_STREQ(error.what(), "10");
-                 }
+               .all([](std::variant<std::runtime_error>&& error) {
+                 EXPECT_STREQ(std::get<0>(error).what(), "10");
+                 return 100;
+               })
+        >> Then([](int value) {
+             return value;
+           });
+  };
+
+  static_assert(
+      eventuals::tuple_types_unordered_equals_v<
+          decltype(e())::ErrorsFrom<void, std::tuple<>>,
+          std::tuple<>>);
+
+  EXPECT_EQ(*e(), 100);
+}
+
+TEST(CatchTest, AllRaisedOneException) {
+  auto e = []() {
+    return Just(500)
+        >> Raise(std::runtime_error("runtime_error"))
+        >> Raise(std::underflow_error("underflow_error"))
+        >> Catch()
+               .raised<std::overflow_error>([](std::overflow_error&& error) {
+                 ADD_FAILURE() << "Encountered unexpected matched raised";
+                 return 10;
+               })
```

Comment:
We need to make this be a bug, you shouldn't be able to try and catch an error that will never propagate to you!